From 7802b89fe48ca0508c63c9f5b17e3fd3f2b3fba6 Mon Sep 17 00:00:00 2001
From: Andra-Teodora Ilie <andra.ilie@nxp.com>
Date: Tue, 15 Feb 2022 17:56:38 +0200
Subject: [PATCH 11/11] s32: plat: Decrease the maximum number of translation
 tables

MAX_XLAT_TABLES should have the smallest value needed to map
the memory regions requested.

Issue: ALB-8461
Upstream-Status: Pending 

Signed-off-by: Andra-Teodora Ilie <andra.ilie@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 plat/nxp/s32/s32_bl2_el3.c                    | 7 -------
 plat/nxp/s32/s32_bl31.c                       | 6 ------
 plat/nxp/s32/s32g/include/s32g_platform_def.h | 6 +++---
 plat/nxp/s32/s32r/include/platform_def.h      | 4 ++--
 4 files changed, 5 insertions(+), 18 deletions(-)

diff --git a/plat/nxp/s32/s32_bl2_el3.c b/plat/nxp/s32/s32_bl2_el3.c
index 22c91ba1f..d97eb2c89 100644
--- a/plat/nxp/s32/s32_bl2_el3.c
+++ b/plat/nxp/s32/s32_bl2_el3.c
@@ -298,13 +298,6 @@ int s32_el3_mmu_fixup(void)
 		MAX_MMAP_REGIONS,
 		"Fewer MAX_MMAP_REGIONS than in s32_mmap will likely result in a MMU exception at runtime");
 
-	_Static_assert(ARRAY_SIZE(s32_mmap) + ARRAY_SIZE(regions) - 1
-#if !defined(PLAT_s32r)
-		+ BL31SRAM_MAX_PAGES
-#endif
-		<= MAX_XLAT_TABLES,
-		"Fewer MAX_XLAT_TABLES than in s32_mmap will likely result in a MMU exception at runtime");
-
 	/* MMU initialization; while technically not necessary, improves
 	 * bl2_load_images execution time.
 	 */
diff --git a/plat/nxp/s32/s32_bl31.c b/plat/nxp/s32/s32_bl31.c
index c4bf9401b..7fc52371c 100644
--- a/plat/nxp/s32/s32_bl31.c
+++ b/plat/nxp/s32/s32_bl31.c
@@ -225,12 +225,6 @@ static void s32_el3_mmu_fixup(void)
 	_Static_assert(ARRAY_SIZE(s32_mmap) + ARRAY_SIZE(regions) - 1 <=
 		MAX_MMAP_REGIONS,
 		"Fewer MAX_MMAP_REGIONS than in s32_mmap will likely result in a MMU exception at runtime");
-	_Static_assert(ARRAY_SIZE(s32_mmap) + ARRAY_SIZE(regions) - 1
-#if !defined(PLAT_s32r)
-	    + BL31SRAM_MAX_PAGES
-#endif
-		<= MAX_XLAT_TABLES,
-		"Fewer MAX_XLAT_TABLES than in s32_mmap will likely result in a MMU exception at runtime");
 
 	/* MMU initialization; while technically not necessary on cold boot,
 	 * it is required for warm boot path processing
diff --git a/plat/nxp/s32/s32g/include/s32g_platform_def.h b/plat/nxp/s32/s32g/include/s32g_platform_def.h
index 655aad149..d1285d69c 100644
--- a/plat/nxp/s32/s32g/include/s32g_platform_def.h
+++ b/plat/nxp/s32/s32g/include/s32g_platform_def.h
@@ -59,12 +59,12 @@
 #if defined IMAGE_BL31
 /* To limit usage, keep these in sync with sizeof(s32_mmap) */
 #define MAX_MMAP_REGIONS		16
-#define MAX_XLAT_TABLES			(MAX_MMAP_REGIONS + BL31SRAM_MAX_PAGES)
+#define MAX_XLAT_TABLES			15
 #endif
 
 #if defined IMAGE_BL2
-#define MAX_MMAP_REGIONS		17
-#define MAX_XLAT_TABLES			(MAX_MMAP_REGIONS + BL31SRAM_MAX_PAGES)
+#define MAX_MMAP_REGIONS        17
+#define MAX_XLAT_TABLES         25
 #endif
 #if defined IMAGE_BL33
 #pragma warning "BL33 image is being built; you should configure it out."
diff --git a/plat/nxp/s32/s32r/include/platform_def.h b/plat/nxp/s32/s32r/include/platform_def.h
index 4c4ec65fc..b97269510 100644
--- a/plat/nxp/s32/s32r/include/platform_def.h
+++ b/plat/nxp/s32/s32r/include/platform_def.h
@@ -1,5 +1,5 @@
 /*
- * Copyright 2021 NXP
+ * Copyright 2021-2022 NXP
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
@@ -22,7 +22,7 @@
 
 #if defined IMAGE_BL2
 #define MAX_MMAP_REGIONS		15
-#define MAX_XLAT_TABLES			25
+#define MAX_XLAT_TABLES			23
 #endif
 
 #if defined IMAGE_BL33
-- 
2.17.1

